### PR TITLE
HAWKULAR-1036 - Check for authorization header value size.

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

**Comment by jpkrohling** [comment]
@pilhuhn , care to review?

File: secret-store-api/src/main/java/org/keycloak/secretstore/api/RequestRewriter.java [hunk 1]
```diff
@@ -94,6 +94,13 @@ public HttpServerExchange rewrite(HttpServerExchange httpServerExchange) throws
 
         String authorization = new String(Base64.getDecoder().decode(authorizationHeaderParts[1]));
         String[] parts = authorization.split(":");
+
+        if (parts.length != 2) {
+            logger.authorizationValueInvalid(authorization);
+            // we don't return a 403 from here, as there might be something down the line that might accept this
```
Comment (pilhuhn) [inline secret-store-api/src/main/java/org/keycloak/secretstore/api/RequestRewriter.java:100]: Do we really want to not reject empty key or password right away?

**Comment by pilhuhn** [comment]
Fine with me - you are the expert :)

File: secret-store-api/src/main/java/org/keycloak/secretstore/api/RequestRewriter.java (hunk 1 above)
Comment (jpkrohling) [inline secret-store-api/src/main/java/org/keycloak/secretstore/api/RequestRewriter.java:100]: Yes, this runs before KC kicks in. So, KC will reject it, unless someone else put another filter which would "accept" this empty credential. In other words: at this point in the code, we are just determining if we are "responsible" for the request.

Comment (pilhuhn) [inline secret-store-api/src/main/java/org/keycloak/secretstore/api/RequestRewriter.java:100]: :+1:

**Pull request merged**